### ISSUE TITLE
"metadata.name: invalid value 'quickstart-java-camel-cdi-service': must have at most 24 characters and match regex [a-z]([-a-z0-9]*[a-z0-9])

### ISSUE THREAD

**Issue by davsclaus** [fabric8io/fabric8#3794]
Trying to start the camel-cdi quickstart app on os1 fails with that error message.

Btw the web console need a "logs notification" center so you can find this error easier. I had to dig it from the javascript console. See screenshot

![screen shot 2015-04-23 at 10 39 33 am](https://cloud.githubusercontent.com/assets/477100/7293370/c61d8fb8-e9a4-11e4-92f9-a70ba01d35ed.png)

**Comment by davsclaus** [comment]
```
[Kubernetes] Failed to save /kubernetes/api/v1beta2/services status: 422 response: {
  "kind": "Status",
  "creationTimestamp": null,
  "apiVersion": "v1beta2",
  "status": "Failure",
  "message": "Service \"quickstart-java-camel-cdi-service\" is invalid: [metadata.name: invalid value 'quickstart-java-camel-cdi-service': must have at most 24 characters and match regex [a-z]([-a-z0-9]*[a-z0-9])?, spec.port: invalid value '0': must be greater than 0 and less than 65536]",
  "reason": "Invalid",
  "details": {
    "id": "quickstart-java-camel-cdi-service",
    "kind": "Service",
    "causes": [
      {
        "reason": "FieldValueInvalid",
        "message": "metadata.name: invalid value 'quickstart-java-camel-cdi-service': must have at most 24 characters and match regex [a-z]([-a-z0-9]*[a-z0-9])?",
        "field": "metadata.name"
      },
      {
        "reason": "FieldValueInvalid",
        "message": "spec.port: invalid value '0': must be greater than 0 and less than 65536",
        "field": "spec.port"
      }
    ]
  },
  "code": 422
}
```

**Comment by jimmidyson** [comment]
Fixed by 11afd3b8a85f15182bb02f496856ca0d58290001
